lib/Launcher.js

```diff
@@ -86,12 +89,16 @@ class Launcher {
       chromeProcess.stderr.pipe(process.stderr);
     }
 
-    const waitForChromeToClose = new Promise(fulfill => {
+    const waitForChromeToClose = new Promise((fulfill, reject) => {
       chromeProcess.once('close', () => {
         // Cleanup as processes exit.
-        if (temporaryUserDataDir)
-          removeSync(temporaryUserDataDir);
-        fulfill();
+        if (temporaryUserDataDir) {
+          removeFolderAsync(temporaryUserDataDir)
```

Comment:
@CheerlessCloud I've seen your comment that the promise is not not catched in a few places, thanks for the clarification.

let's fulfill regardless of the temprorary data directory cleanup, and use `debugError` instead of `console.error`:

```js
if (temporaryUserDataDir) {
  removeFolderAsync(temporaryUserDataDir)
    .catch(debugError)
    .then(fulfill);
} else {
  fulfill();
}
```

Reply:
I want to be able to try/catch the error that removing the temporaryUserDataDir throws. You should await it and throw.

Reply:
@graingert what are you going to do with the exception? The `temporaryUserDataDir` is internal to launcher's implementation.